l2l/optimizers/kalmanfilter/kalman_utils.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+from numpy.linalg import norm, inv
+from numpy import sqrt
+
+
+def _get_mean(x):
+    """
+    Depending on the shape returns the correct mean
+    """
+    if x.shape[1] == 1:
+        return np.mean(x)
+    return np.mean(x, axis=0)
+
+
+def _get_shapes(observations, model_output):
+    """
+    Returns individual shapes
+
+    :returns gamma_shape, length of the observation (here: size of last layer
+                          of network)
+    :returns dimensions, number of observations (and data)
+    """
+    if model_output.ndim > 2:
+        gamma_shape = model_output.shape[1]
+    else:
+        gamma_shape = model_output.shape[0]
+    dimensions = observations.shape[0]
+    return gamma_shape, dimensions
+
+
+def _calculate_cost(y, y_hat, loss_function='BCE'):
+    """
+    Loss functions
+    :param y: target
+    :param y_hat: calculated output (here G(u) or feed-forword output a)
+    :param loss_function: name of the loss function
+           `MAE` is the Mean Absolute Error or l1 - loss
+           `MSE` is the Mean Squared Error or l2 - loss
+           `CE` cross-entropy loss, requires `y_hat` to be in [0, 1]
+           `norm` norm-2 or Forbenius norm of `y - y_hat`
+    :return: cost calculated according to `loss_function`
+    """
+    if loss_function == 'CE':
```

Comment:
Could you make this an enum instead of string?

Reply:
Can you motivate why `enum` should be used?

Reply:
To prevent bugs that result from misspelling in strings and to allow linting tools to be able to check this programmatically. If I'm not mistaken, your code has this very issue currently, where the default argument is 'BCE', which would lead to a `KeyError` since it's not handled in the if-else statements below. If it had been Enums, your linting tool would have caught it. :-)